src/System.Sorting/tests/SortingTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using Microsoft.Xunit.Performance;
+
+
+namespace System.Sorting.Tests
+{
+    public class SortingTest
+    {
+        [Fact]
+        public static void TestTimSort()
+        {
+            IComparer<int> iComparer= new IntegerComparer();
+
+            //int[] array = new int[] { 67, 68, 69, 70, 71, 72, 10, 8, 7, 6, 5, 20, 18, 536, 537, 538, 22, 44, 74, 90, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119 };
+            int[] array = new int[] { 67, 68, 69, 70, 71, 72, 10, 8, 7, 6, 5, 20, 18, 536, 537, 538, 22, 44, 74, 90, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119,22,21,20,19,18,49,48,47,46 };
+            int[] arrayClone = (int[])(array.Clone());
+
+            TimSort<int> timsort = new TimSort<int>(array, iComparer);
+            timsort.Sort(array, 0, array.Length, iComparer);           
+            //TimSort<int>.Default(array, iComparer).Sort(array, 0, array.Length, iComparer);
+            Array.Sort(arrayClone, 0, arrayClone.Length, iComparer);           
```

Comment:
Instead of doing Array.Sort here, I would just loop over the elements and use the `IComparer` to make sure that the `i`'th element is less than or equal to the `i+1`'th element.